targets/posix/nanoCLR/CLRStartup.cpp

```
@@ -0,0 +1,544 @@
+            delete it.second;
+        m_assemblies.clear();
+    }
+};
+
+static Settings s_ClrSettings;
+
+// ── Helper: wchar_t* → std::string ─────────────────────────────────────────
+// Assembly names are always ASCII; just mask the high bits.
+// On POSIX wchar_t is 4-byte; on Windows the caller passes UTF-16 (2-byte)
+// data into a wchar_t* parameter.  Either way the low byte gives us ASCII.
+static std::string WcharToString(const wchar_t *w)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)